src/sentry/api/serializers/models/alert_rule.py

```diff
@@ -127,23 +135,33 @@ def get_attrs(
                     ).get("uuid")
             alert_rule_triggers.append(serialized)
 
-        alert_rule_projects = set()
-        for alert_rule in alert_rules.values():
+        alert_activations_ranked = AlertRuleActivations.objects.annotate(
+            rank=Window(
+                expression=RowNumber(),
+                partition_by=[F("alert_rule_id")],
+                order_by=F("date_added").desc(),
+            )
+        )
+        activations = alert_activations_ranked.filter(alert_rule__in=item_list, rank__lte=10)
+        activations_by_alert_rule_id = defaultdict(list)
+        for activation in activations:
+            activations_by_alert_rule_id[activation.alert_rule_id].append(activation)
+
+        # Construct list of all projects associated with each alert rule
+        # given both the project fk as well as the snuba query projects relationship
+        projects_per_alert_rule = defaultdict(set)
+        for alert_rule in item_list:
             try:
-                alert_rule_projects.add([alert_rule.id, alert_rule.projects.get().slug])
+                projects_per_alert_rule[alert_rule].add(alert_rule.projects.get().slug)
+                snuba_query_project = (
+                    alert_rule.snuba_query.get().subscriptions.get().project.get().slug
+                )
+                projects_per_alert_rule[alert_rule.id].add(snuba_query_project)
             except Exception:
                 pass
```

Comment:
This should be done in bulk, there's a big n+1 problem right now. We should also not be catching `Exception` here and just passing